workspaces/src/rpc/client.rs

```diff
@@ -559,22 +558,13 @@ pub(crate) async fn send_batch_tx_and_retry(
     receiver_id: &AccountId,
     actions: Vec<Action>,
 ) -> Result<FinalExecutionOutcomeView> {
-    let signer = signer.inner();
-    let cache_key = (signer.account_id.clone(), signer.public_key());
-
+    let cache_key = (signer.account_id.clone(), signer.secret_key.public_key().0);
     retry(|| async {
         let (block_hash, nonce) = fetch_tx_nonce(client, &cache_key).await?;
         send_tx(
             client,
             &cache_key,
-            SignedTransaction::from_actions(
-                nonce,
-                signer.account_id.clone(),
-                receiver_id.clone(),
-                &signer as &dyn Signer,
-                actions.clone(),
-                block_hash,
-            ),
+            signed_transaction(signer, actions.clone(), nonce, block_hash, receiver_id),
```

Comment:
I prefer constructor method (`from_actions` or maybe other name) on SignedTransaction over a standalone function as it is harder to discover the function. What is exactly the motivation behind this change?

Reply:
The function is a reimplementation of `from_actions` without requiring `Signer trait` as an argument. The hint I took [here](https://github.com/near/near-workspaces-rs/issues/153) is that we can condense that implementation and remove the reliance on the trait.
I could name it better.